### PR TITLE
Set bit

### PR DESCRIPTION
**Pull request by lukasb1b**
- [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Java/blob/master/CONTRIBUTING.md).
- [x] This pull request is all my own work -- I have not plagiarized it.
- [x] All filenames are in PascalCase.
- [x] All functions and variable names follow Java naming conventions.
- [x] All new algorithms have a URL in their comments that points to Wikipedia or other similar explanations.

### CODE REVIEW

**Comment by lukasb1b** [comment]
Issue #4350

**Comment by lukasb1b** [comment]
I want to add to the Issue that SetBit is already implemented and ClearBit is in other pullrequest

**Comment by siriak** [comment]
@vil02 @BamaCharanChhandogi please review

**Comment by BamaCharanChhandogi** [comment]
> I want to add to the Issue that SetBit is already implemented and ClearBit is in other pullrequest

Hi, @lukasb1b,
All functions of single bit manipulation should be included in one file named 'Single Bit Manipulation,'. That would be better.

**Comment by vil02** [comment]
@siriak, @BamaCharanChhandogi, @lukasb1b: please have a look at [my comment in the corresponding issue](https://github.com/TheAlgorithms/Java/issues/4350#issuecomment-1710635231).

File: src/main/java/com/thealgorithms/bitmanipulation/GetBit.java [hunk 1]
```diff
@@ -0,0 +1,10 @@
+package com.thealgorithms.bitmanipulation;
+/**
+ * Returns the value of bit from num located at get
+ */
+
+public class GetBit {
```
Comment (vil02) [inline src/main/java/com/thealgorithms/bitmanipulation/GetBit.java:6]: This is a very good start for the [`BitArray` class](https://github.com/TheAlgorithms/Java/issues/4350#issuecomment-1710635231).

**Review by vil02 — CHANGES_REQUESTED** [review]

File: src/main/java/com/thealgorithms/bitmanipulation/GetBit.java (hunk 1 above)
Comment (siriak) [reply src/main/java/com/thealgorithms/bitmanipulation/GetBit.java:6]: I suggest adding this as-is and consolidating algorithms into a data structure in a following PR. There are already some other functions implemented in this manner and, as you have mentioned, there are open questions about the design of this BitArray.
@lukasb1b will you be able to consolidate these bit functions into a BitArray in the next PR? If no, somebody else could pick up this task.
In the meantime, we can discuss what we want to use as an underlying type.

File: src/main/java/com/thealgorithms/bitmanipulation/GetBit.java [hunk 2]
```diff
@@ -0,0 +1,10 @@
+package com.thealgorithms.bitmanipulation;
+/**
+ * Returns the value of bit from num located at get
+ */
+
+public class GetBit {
+    public static int getBit(int num, int get) {
+        return (num >> get) & 1;
+    }
+}
```
Comment (vil02) [inline src/main/java/com/thealgorithms/bitmanipulation/GetBit.java:10]: In any case such functionality will be needed.

```suggestion
public final class GetBit {
    private GetBit() {
    }
    public static boolean get(final int data, final int pos) {
        if (pos < 0 || pos >= Integer.SIZE) {
            throw new ArrayIndexOutOfBoundsException();
        }
        return ((data >> pos) & 1) > 0;
    }
}
```

I am not sure what is the best way to make it accept any `IntegralType` - I could not find a convenient way how to get the `SIZE`.

Please add tests for the missing _branches_.

**Review by vil02 — CHANGES_REQUESTED** [review]

File: src/main/java/com/thealgorithms/bitmanipulation/GetBit.java (hunk 2 above)
Comment (siriak) [reply src/main/java/com/thealgorithms/bitmanipulation/GetBit.java:10]: Let's stick to int32 then

**Comment by lukasb1b** [comment]
I can add all the SingleBit functions in onte but at first i think we should have a solution for the Problem i personally think a bit array is not as functional but we could do it generic like in the example below. And we add another function for arrays.

public class GetBit<T extends Number>{

    public static <T extends Number> T getBit(T num, int get) throws Exception {
        if(num instanceof Integer){
            return (T)(Integer.valueOf((num.intValue()>>get) & 1));
        }else{
            throw new Exception();
        }
    }

**Comment by vil02** [comment]
> I can add all the SingleBit functions in onte but at first i think we should have a solution for the Problem i personally think a bit array is not as functional but we could do it generic like in the example below. And we add another function for arrays.
> 
> public class GetBit{
> 
> ```
> public static <T extends Number> T getBit(T num, int get) throws Exception {
>     if(num instanceof Integer){
>         return (T)(Integer.valueOf((num.intValue()>>get) & 1));
>     }else{
>         throw new Exception();
>     }
> }
> ```

As [suggested before](https://github.com/TheAlgorithms/Java/pull/4356#discussion_r1319772921), let stick to `int` for now. If you agree with [my suggestion](https://github.com/TheAlgorithms/Java/pull/4356#discussion_r1319405642), please apply it and add missing tests.

I agree that it is a good idea to have one _utilitiy class_ like `BitOperations` with `public static` methods `get` and `set` (this will be needed anyway in the `BitArray`).

**Comment by lukasb1b** [comment]
Hi I closed the request so you can implement it I dont get how you want to do it so I think its quicker if you do it yourself

**Pull request closed without merging**